pkg/policy/policy_controller.go

```diff
@@ -635,6 +691,54 @@ func getTrigger(rd kyverno.ResourceDescription) []*kyverno.ResourceSpec {
 	return specs
 }
 
+func generateTriggers(lister listerv1.NamespaceLister, rule kyverno.Rule, log logr.Logger) []*kyverno.ResourceSpec {
+	var specs []*kyverno.ResourceSpec
+
+	nslist, err := lister.List(labels.NewSelector())
+	if err != nil {
+		log.Error(err, "failed to list namespaces")
+	}
+
+	triggers := genTrigger(nslist, rule.MatchResources.ResourceDescription)
+	specs = append(specs, triggers...)
+
+	for _, any := range rule.MatchResources.Any {
+		triggers := genTrigger(nslist, any.ResourceDescription)
+		specs = append(specs, triggers...)
+	}
+
+	triggers = []*kyverno.ResourceSpec{}
+	for _, all := range rule.MatchResources.All {
+		triggers = genTrigger(nslist, all.ResourceDescription)
```

Comment:
This was not used?

It will be logic AND for `match.all`, you need to get the intersect.
```
	// All allows specifying resources which will be ANDed
	// +optional
	All ResourceFilters `json:"all,omitempty" yaml:"all,omitempty"`
```